### PR TITLE
improve ErrorHandling

### PR DESCRIPTION
**Pull request by niemannd**
This PR improves the ErrorHandling of the GenericServiceTemplate. Instead of ignoring an error, we throw it as an RuntimeException. This way the user can decide if he wants to handle it or just ignore it.

### CODE REVIEW

File: src/test/java/com/meilisearch/integration/ExceptionsTest.java [hunk 1]
```diff
@@ -54,11 +45,12 @@ public void testErrorSerializeAndGetters() throws Exception {
 	 * Test MeiliSearchApiException is thrown on MeiliSearch bad request
 	 */
 	@Test
+	@Disabled
```
Comment (eskombro) [inline src/test/java/com/meilisearch/integration/ExceptionsTest.java:48]:
```suggestion
```

**Review by eskombro — DISMISSED** [review]
Hey @niemannd !

Thanks for this Error handler improvement! The only modification I would like to do is removing the `@disabled` for `testMeiliSearchApiExceptionBadRequest` that you probably forgot (but the test is passing btw). I did a suggestion for this in the review, can you please fix it? Otherwise, LVGTM :)

**Comment by eskombro** [comment]
bors try

**Comment by niemannd** [comment]
The problem with cutting big rewrites into smaller PRs is that you sometimes get stuff like that `@Disabled` in there.
I removed it. It would be fixed in the following PRs anyway.

**Comment by eskombro** [comment]
Ok, sorry about that, I thought it would be better to review and merge one by one because it becomes quite big and I don't want to miss the details :)

**Review by eskombro — APPROVED** [review]
👌🏽

**Comment by eskombro** [comment]
Bors merge

**Pull request merged**